File: IntegrationGuideForAndroid/Apps/frameworkExtendJava/src/main/java/com/example/integrationguide/AWApplication.java
// Copyright 2023 VMware, Inc.
// SPDX-License-Identifier: BSD-2-Clause

package com.example.integrationguide;

import android.content.Intent;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatDelegate;

import com.airwatch.event.WS1AnchorEvents;

import org.jetbrains.annotations.NotNull;

import java.security.cert.X509Certificate;

// Note the fully qualified class name in the extends declaration.
public class AWApplication extends com.airwatch.app.AWApplication {
    @NotNull
    @Override
    public Intent getMainActivityIntent() {
        return new Intent(getApplicationContext(), MainActivity.class);
    }

    @Override
    public void onSSLPinningRequestFailure(
            @NotNull String host, X509Certificate x509Certificate
    ) {
    }

    @Override
    public void onSSLPinningValidationFailure(
            @NotNull String host, X509Certificate x509Certificate
    ) {
    }

    @Override
    public int getNightMode() {
        return AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
    }

    @NonNull
    @Override
    public WS1AnchorEvents getEventHandler() {
        return new AppWS1AnchorEvents();
    }
}

File: IntegrationGuideForAndroid/Apps/tunnel/src/main/java/com/example/integrationguide/AWApplication.kt
// Copyright 2023 VMware, Inc.
// SPDX-License-Identifier: BSD-2-Clause

package com.example.integrationguide

import android.content.Intent
import androidx.appcompat.app.AppCompatDelegate
import com.airwatch.event.WS1AnchorEvents
import java.security.cert.X509Certificate

// Note the fully qualified base class name.
open class AWApplication: com.airwatch.app.AWApplication() {
    override fun getMainActivityIntent(): Intent {
        return Intent(applicationContext, MainActivity::class.java)
    }

    override fun onSSLPinningRequestFailure(
        host: String,
        serverCACert: X509Certificate?
    ) {
    }

    override fun onSSLPinningValidationFailure(
        host: String,
        serverCACert: X509Certificate?
    ) {
    }

    override fun getNightMode(): Int {
        return AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
    }

    override fun getEventHandler(): WS1AnchorEvents {
        return AppWS1AnchorEvents()
    }
}
